substrate-archive/src/simple_db.rs

```diff
@@ -1,143 +0,0 @@
-// Copyright 2017-2019 Parity Technologies (UK) Ltd.
-// This file is part of substrate-archive.
-
-// substrate-archive is free software: you can redistribute it and/or modify
-// it under the terms of the GNU General Public License as published by
-// the Free Software Foundation, either version 3 of the License, or
-// (at your option) any later version.
-
-// substrate-archive is distributed in the hope that it will be useful,
-// but WITHOUT ANY WARRANTY; without even the implied warranty of
-// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-// GNU General Public License for more details.
-
-// You should have received a copy of the GNU General Public License
-// along with substrate-archive.  If not, see <http://www.gnu.org/licenses/>.
-
-//! a simple "database" that serializes to/from bincode
-//! used for saving entries which aren't ready to insert into the database (because of E.G missing blocks)
-//! and saves these entries to disk to be reloaded the next time a program starts
-//! avoids having to search for missing entries in PostgreSQL and re-fetching from running node
```

Comment:
Can you elaborate a bit more on why you think this functionality is not useful?

Reply:
The `simple_db` module is not used by any other modules, nor is it re-exported from the library, so I think it's useless, maybe someone forgot to delete it before

Reply:
I see, hmm. @insipx can you help us clarify the status of `simple_db`?

Reply:
simple_db was used for tests long ago, I never got rid of it because I thought it might still be useful for tests at some point, then I forgot about it. Totally OK to delete it

Reply:
Great, good find @koushiro.

@insipx Can we do something to stop dead code from lingering in the code base going forwards? I think test coverage can be improved for `archive`, but maybe there's low hanging fruit to pick in CI, `[deny(dead_code)]` or some such?

Reply:
Sure, I think deny(dead_code) would be good, made #199 for it. Hopefully that should help the situation